src/Nethermind/Nethermind.Optimism/OptimismBaseFeeCalculator.cs

```diff
@@ -0,0 +1,35 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Int256;
+
+namespace Nethermind.Optimism;
+
+/// <remarks>
+/// See <see href="https://specs.optimism.io/protocol/holocene/exec-engine.html#base-fee-computation"/>
+/// </remarks>
+public sealed class OptimismBaseFeeCalculator(
+    IBaseFeeCalculator baseFeeCalculator,
+    ISpecProvider specProvider) : IBaseFeeCalculator
```

Comment:
I don't see much point in having `IOptimismSpecHelper` for starters, but could replace this `ISpecProvider` with `IOptimismSpecHelper`, and replace `releaseSpec.GetSpec(parent).IsOpHoloceneEnabled` with `helper.IsHolocene(parent)`.

If we don't use `IOptimismSpecHelper` then its additions related to Holocene are essentially dead code (no usages).